GVRf/Framework/src/org/gearvrf/GVRViewManager.java

```diff
@@ -246,27 +260,154 @@ void beforeDrawEyes() {
+            int width = 0, height = 0;
+            if (mScreenshotCenterCallback != null
+                    || mScreenshotLeftCallback != null
+                    || mScreenshotRightCallback != null) {
+                GVRRenderTexture rendernTexture = mRenderBundle.getRightRenderTexture();
+                width = rendernTexture.getWidth();
+                height = rendernTexture.getHeight();
+                readbackBuffer = ByteBuffer.allocateDirect(width * height * 4);
+                readbackBuffer.order(ByteOrder.nativeOrder());
+            }
+
             if (eye == 1) {
                 mainCameraRig.predict(4.0f / 60.0f);
                 GVRCamera rightCamera = mainCameraRig.getRightCamera();
                 renderCamera(mActivity.appPtr, mMainScene, rightCamera,
                         mRenderBundle.getRightRenderTexture(), mRenderBundle);
+
+                // if mScreenshotRightCallback is not null, capture right eye
+                if (mScreenshotRightCallback != null) {
+                    readRenderResult(mRenderBundle
+                            .getPostEffectRenderTextureA().getPtr(),
+                            readbackBuffer);
+                    returnScreenshotToCaller(mScreenshotRightCallback,
+                            readbackBuffer, width, height);
+
+                    mScreenshotRightCallback = null;
+                }
+
                 mActivity.setCamera(rightCamera);
             } else {
                 mainCameraRig.predict(3.5f / 60.0f);
+
+                // if mScreenshotCenterCallback is not null, capture center eye
+                if (mScreenshotCenterCallback != null) {
+
+                    // temporarily create a center camera
+                    GVRCamera centerCamera = new GVRPerspectiveCamera(this);
+                    centerCamera.setRenderMask(GVRRenderMaskBit.Left
+                            | GVRRenderMaskBit.Right);
+                    GVRSceneObject centerCameraObject = new GVRSceneObject(this);
+                    centerCameraObject.attachCamera(centerCamera);
+                    mainCameraRig.getOwnerObject().addChildObject(
+                            centerCameraObject);
+
+                    renderCamera(mActivity.appPtr, mMainScene, centerCamera,
+                            mRenderBundle.getRightRenderTexture(),
+                            mRenderBundle);
+                    
+                    centerCameraObject.detachCamera();
+                    mainCameraRig.getOwnerObject().removeChildObject(
+                            centerCameraObject);
+
+                    readRenderResult(mRenderBundle
+                            .getPostEffectRenderTextureA().getPtr(),
+                            readbackBuffer);
+                    returnScreenshotToCaller(mScreenshotCenterCallback,
+                            readbackBuffer, width, height);
+
+                    mScreenshotCenterCallback = null;
+                }
```

Comment:
In fact, even if we do not predict at all (i.e. passing 0 to predict), we would still have the same problem.

This is because we render left eye and right at different time, the head orientation may be different. The better way should be render both eyes in one function call, but it is out of our control.

In the current implementation, the center camera already has spacial interpolation between left and right cameras. But it lacks temporal interpolation. We may store the orientation of the left camera when we render left eye, and use the stored value when we render right eye to compute the "temporal interpolation". This will of course lead to more computations.

I think the current spacial interpolation should be good enough for this screen shot.